src/core/server/saved_objects/service/lib/errors.ts

```diff
@@ -202,4 +202,23 @@ export class SavedObjectsErrorHelpers {
   public static isGeneralError(error: Error | DecoratedError) {
     return isSavedObjectsClientError(error) && error[code] === CODE_GENERAL_ERROR;
   }
+
+  public static createGenericNotFoundEsUnavailableError(
+    type: string | null = null,
```

Comment:
under what circumstances can they be `null`?

Reply:
Good question! This PR only adds the checks to `get`, `update` and `delete`, all of which require a `type` and `id`.

Reply:
[done](https://github.com/elastic/kibana/pull/107301/files#diff-7a19c0f1ba03e2b736d8031dc6c5ea8b746cae37d2d81f357577ec34612dbc12R206)